addons/mail/models/mail_activity.py

```diff
@@ -308,6 +321,16 @@ def action_done_schedule_next(self):
     def action_close_dialog(self):
         return {'type': 'ir.actions.act_window_close'}
 
+    @api.multi
+    def activity_format(self):
+        activities = self.read()
```

Comment:
Maybe we could filter what we read and avoid reading mail_template_ids as we read them just below.

Reply:
I used mapped by laziness and forgot to change that, we can get the template_ids from from activities
mail_template_ids = set([template_id for activity in activities for template_id in activity["mail_template_ids"]])

Do you think with should filter the fields we read anyway? 
+Maybe a little lighter
-Less flexible
-Readability

Reply:
Ok I melanged with the m2o.